go/pkg/cas/ioutil.go

```diff
@@ -1,39 +1,85 @@
 package cas
 
 import (
+	"bufio"
 	"bytes"
 	"fmt"
 	"io"
+	"os"
+	"sync"
 )
 
 // This file contains various IO utility types and functions.
 
-// readSeekCloser can Read, Seek and Close.
-// TODO(nodir): use io.ReadSeekCloser after dropping Go 1.15 support.
-type readSeekCloser interface {
-	io.ReadSeeker
+// byteSource is a common interface to in-memory blobs and files.
+// SeekStart is like io.Seeker.Seek, but only supports io.SeekStart.
+type byteSource interface {
```

Comment:
`source` instead of `byteSource`? I think `byteSource` can imply bytes in memory.

Reply:
but it is a source of bytes :) How does `uploadSource` sound? `source` is a bit too generic.

Reply:
heh, `uploadSource` sgtm. Maybe it's just me but `byte` reminds me of `[]byte` :)